packages/core/src/Utils/Analytics/index.ts

```diff
@@ -49,10 +50,14 @@ export const AnalyticsInitializer = async () => {
                         network_type: navigator.connection?.effectiveType,
                         network_rtt: navigator.connection?.rtt,
                         network_downlink: navigator.connection?.downlink,
+                        user_id: localStorage.getItem('active_user_id') || '',
                     },
                 },
             };
             await Analytics?.initialise(config);
+            const anonymousId = Analytics.getInstances()?.tracking?.getAnonymousId?.();
+            config.growthbookOptions.attributes.anonymous_id = anonymousId;
+            Analytics?.initialise(config);
```

Comment:
Also as per analytics package code - 
`  growthbookOptions.attributes.id ??= _rudderstack.getAnonymousId()`
we are setting ` rudderstack-anonymousId` to "id" .   
- why do we need both `anonymous_id` and `id`  attribute in growthbook?  Both are mapped to rudderstack-anonymousId
<img width="877" alt="Screenshot 2024-12-25 at 12 45 15 PM" src="https://github.com/user-attachments/assets/6a7e4ded-16f8-4655-8c13-12073d2de63a" />

- Cant we handle this mapping in analytics package so that we can remove reinitialize?

Reply:
Yes we will be removing the id and replacing it with anonymous_id